src/main/java/appeng/tile/storage/TileIOPort.java

```diff
@@ -277,87 +276,84 @@ public TickRateModulation tickingRequest( final IGridNode node, final int ticksS
 			return TickRateModulation.IDLE;
 		}
 
-		long ItemsToMove = 256;
+		TickRateModulation ret = TickRateModulation.SLEEP;
+		long itemsToMove = 256;
 
 		switch( this.getInstalledUpgrades( Upgrades.SPEED ) )
 		{
 			case 1:
-				ItemsToMove *= 2;
+				itemsToMove *= 2;
 				break;
 			case 2:
-				ItemsToMove *= 4;
+				itemsToMove *= 4;
 				break;
 			case 3:
-				ItemsToMove *= 8;
+				itemsToMove *= 8;
 				break;
 		}
 
 		try
 		{
-			final IMEInventory<IAEItemStack> itemNet = this.getProxy().getStorage().getInventory(
-					AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) );
-			final IMEInventory<IAEFluidStack> fluidNet = this.getProxy().getStorage().getInventory(
-					AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
 			final IEnergySource energy = this.getProxy().getEnergy();
-			for( int x = 0; x < 6; x++ )
+			for( int x = 0; x < NUMBER_OF_CELL_SLOTS; x++ )
 			{
 				final ItemStack is = this.inputCells.getStackInSlot( x );
 				if( !is.isEmpty() )
 				{
-					if( ItemsToMove > 0 )
-					{
-						final IMEInventory<IAEItemStack> itemInv = this.getInv( is, AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) );
-						final IMEInventory<IAEFluidStack> fluidInv = this.getInv( is,
-								AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
+					boolean shouldMove = true;
 
-						if( this.manager.getSetting( Settings.OPERATION_MODE ) == OperationMode.EMPTY )
+					for( IStorageChannel<? extends IAEStack<?>> c : AEApi.instance().storage().storageChannels() )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)